cli/README.md

```diff
@@ -224,6 +224,108 @@ model User {
+    fieldPatterns: {
+      "@updatedAt": "/^updatedAt|updated_at$/",
+      "@createdAt": "/^createdAt$/",
+      "@id @default(cuid())": "/_id$/",
+    }
+  }
+}
+```
+
+> Note: I am not wild about the format itself, but it can be changed in the future
+
+See the full Prisma Config spec for more details
+
+If one or more Introspection Configuration patterns are applied during introspection, the CLI output is amended to include an informational line:
+
+```
+Introspection Configuration patterns were applied 7 times. See [link to docs] for details
+```
+
+The docs should explain how Introspection configuration works, including how to disable it.
+
+### Inferred Introspection Configuration
+
+If there is no `prismarc.json` file, `prisma introspect` will run a set of heuristics to identify common patterns. If one or more patterns are identified, a `prismarc.json` file is created, and all the patterns are written to it. `prisma introspect` will then proceed as normally, and take the newly created Introspection Configuration into account.
+
+The common patterns are pciked to minimise false positives while providing an easy transition from most popular ORMs, including Sequelize, TypeORM, Objection, ActiveRecord, Hibernate, Laravel. Additionally, it is a goal to provide a smooth upgrade path for Prisma1 users.
+
+List of inferred patterns:
+
+> Note: This list should be moved to a separate introspection spec
+
+**Variations of updatedAt fields**
+
+If a column has any of the names `updatedAt`, `updated_at`, `updated_on`, `updated`, `updateDate`, `updatedDate`, `lastModifiedDate` and is of a type that maps to the Prisma type `DateTime`, the following fieldPattern is added:
+
+```
+"@updatedAt": "/^updatedAt|updated_at|updated_on|updated|updateDate|updatedDate|lastModifiedDate$/",
+```
+
+**Variations of createdAt fields**
+
+If a column has any of the names `createdAt`, `created_at`, `created_on`, `created`, `createDate`, `createdDate` and is of a type that maps to the Prisma type `DateTime`, the following fieldPattern is added:
+
+```
+"@createdAt": "/^createdAt|created_at|created_on|created|createDate|createdDate$/",
+```
+
+**UUID**
+
+If a column has the name `id` and native database type `UUID` or `CHAR(36)` or `VARCHAR(36)`, the following pattern is added:
+
+```
+"@id @default(uuid())": "/^id$/",
+```
+
+**CUID**
+
+If a column has the name `id` and native database type `VARCHAR(24)`, the following pattern is added:
+
+```
+"@id @default(cuid())": "/^id$/",
+```
+
+### Disabling Introspection Configuration
+
+Introspection Configuration and Inferred Introspection Configuration can be disavled by leaving the `prismarc.json` empty, or set `"fieldPatterns"` to an empty object.
+
+### Considerations for the first-run experience
+
+When a user first approaches Prisma, they will rin `prisma generate` immediadtely after running `prisma init`. As such, there will be no `prismarc.json` file, an the Inferred Introspection Configuration is triggered.
```

Comment:
"they will run `prisma generate`" typo.

I would like to mention first thing I ran after init was `prisma introspect`, as this was the biggest selling point for me.

Shoudnt the `prisma init` generate the configuration file too?